### ISSUE TITLE
False positive on `enum_clike_unportable_variant`

### ISSUE THREAD

**Issue by llogiq** [Manishearth/rust-clippy#816]
```
x = 1isize << 31; 
```
is okay as long as the specific bit pattern is of the essence; the actual value may vary.

E.g. see https://github.com/ledbettj/wavefile/blob/master/src/speakers.rs#L20

**Comment by oli-obk** [comment]
shouldn't the type in the example rather be `i32`?

**Comment by oli-obk** [comment]
if you want the bit pattern in a portable way use `(1usize << 31) as isize`, which is more explicit to your intent

**Comment by llogiq** [comment]
You are technically correct (which is after all the best kind of correct :smile:). Still I think it may be useful to split the lint so we can give more relevant report messages.
